addons/addon-base-raas/packages/base-raas-services/lib/data-egress/data-egress-service.js

```diff
@@ -124,15 +138,114 @@ class DataEgressService extends Service {
         },
       ],
     };
+    const memberAccountId = await this.getMemberAccountId(requestContext, environment.id);
 
-    const environmentScService = await this.service('environmentScService');
-    const environmentScEntity = await environmentScService.mustFind(requestContext, { id: environment.id });
-    const memberAccount = await environmentScService.getMemberAccount(requestContext, environmentScEntity);
-    await this.addEgressStoreBucketPolicy(requestContext, egressStore, memberAccount.accountId);
+    const lockService = await this.service('lockService');
+    const lockId = `bucket-policy-access-${bucketName}`;
+    await lockService.tryWriteLockAndRun({ id: lockId }, async () => {
+      await this.addEgressStoreBucketPolicy(requestContext, egressStore, memberAccountId);
+    });
 
     return egressStore;
   }
 
+  async terminateEgressStore(requestContext, environmentId) {
+    const enableEgressStore = this.settings.get(settingKeys.enableEgressStore);
+    const curUser = _.get(requestContext, 'principalIdentifier.uid');
+    if (!enableEgressStore) {
+      throw this.boom.forbidden('Unable to create Egress store since this feature is disabled', true);
+    }
+
+    const workspaceId = environmentId;
+    let egressStoreScanResult = [];
+
+    try {
+      egressStoreScanResult = await this._scanner()
+        .limit(1000)
+        .scan()
+        .then(egressStores => {
+          return egressStores.filter(store => store.workspaceId === workspaceId);
+        });
+    } catch (error) {
+      throw this.boom.notFound(`Error in terminating egress store: ${JSON.stringify(error)}`, true);
+    }
+    const egressStoreInfo = egressStoreScanResult[0];
```

Comment:
probably an error if you see more than one result

Reply:
Fixed in commit: c33f4053406ffd30ff50181448e5cba7a7925a90